optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/impl/ConsecutiveSetTree.java

```diff
@@ -252,8 +255,8 @@ public boolean remove(ValueType_ item) {
         ValueType_ endItem = bag.getLastItem();
         itemSet.remove(item);
 
-        // Count of item in bag is 0
-        if (bag.isEmpty()) {
+        // Bag is empty if first item = last item
+        if (bag.getFirstItem() == bag.getLastItem()) {
```

Comment:
Strictly speaking - if the bag's first and last item is the same item, the bag has 1 item inside.
If the first and last item are null, then I could see how the bag is empty.

Reply:
This is called in the remove method, after an item is removed. 1 - 1 = 0.